### PR TITLE
RDK-34568: Expand AVInput RDK Service to include HDMIInput and Compos…

### PR DESCRIPTION
**Pull request by dhivya-ilangovan**
…iteInput APIs

Reason for change: Combine HdmiInput and CompositeInput Plugins into AVInput
Test Procedure: sanity testases on HDMI and COMPOSITE inputs
Risks: Low
Signed-off-by: Dhivya Ilangovan <dhivya.dilangovan@sky.uk>

### CODE REVIEW

File: AVInput/AVInput.cpp [hunk 1]
```diff
@@ -18,193 +18,903 @@
...
+                    break;
+            }
+            sendNotify(AVINPUT_EVENT_ON_SIGNAL_CHANGED, params);
+        }
 
-    auto result = numberOfInputs(success);
-    if (success) {
-        response[_T("numberOfInputs")] = result;
-    }
+        /**
```
Comment (ddevad) [inline AVInput/AVInput.cpp:487]: Comment needs to be updated.

Comment (ddevad) [reply AVInput/AVInput.cpp:487]: To maintain consistency , please add for all APIs

**Comment by Firdose2109** [comment]
Can one of the admins verify this patch?

File: AVInput/AVInput.cpp (hunk 1 above)
Comment (dhivya-ilangovan) [reply AVInput/AVInput.cpp:487]: Updated the comments

**Comment by npoltorapavlo** [comment]
This PR breaks unit tests. Also, it reverts #2372. If you want to add new features this is possible without reverts. fyi, AbstractPlugin and utils have been removed from AVInput. Readme was removed as documentation is at https://rdkcentral.github.io/rdkservices/#/api/AVInputPlugin. AVInput has [unit tests](https://github.com/rdkcentral/rdkservices/blob/sprint/2203/test/tests/AVInputTest.cpp), which should be maintained. cc: @anand-ky

**Review by anand-ky — CHANGES_REQUESTED** [review]
@dhivya-ilangovan - Pls check Nikita's comments and help fix the unit tests.

**Comment by anand-ky** [comment]
Closing older PRs. Pls resubmit to latest 2205 sprint.

**Pull request closed without merging**